deployment/docker-compose.yml

```diff
@@ -11,28 +14,31 @@ services:
     ports:
       - 9000:9000
     command: --config /config.yaml
+
   frontend:
     image: "ghcr.io/serokell/edna-frontend:latest"
     depends_on:
       - backend
     ports:
-      - 8080:80
+      - ${FRONTEND_PORT}:80
```

Comment:
Does it work without `env_file` for frontend? How does it know `FRONTEND_PORT`?

Reply:
@worm2fed ping.

Reply:
@gromakovsky oh, I thought I answered.
https://github.com/serokell/edna/pull/57#issuecomment-824816992

it's working

Reply:
<img width="521" alt="image" src="https://user-images.githubusercontent.com/6579275/116056036-01be2880-a686-11eb-94d1-8cc3e64251f2.png">

<img width="252" alt="image" src="https://user-images.githubusercontent.com/6579275/116056052-0682dc80-a686-11eb-8364-c9fc0bd3763c.png">

<img width="457" alt="image" src="https://user-images.githubusercontent.com/6579275/116056081-0da9ea80-a686-11eb-94bb-8b87b06bcfa0.png">

Reply:
Okay, but how does it know above `.env`? We don't specify `env_file` for `frontend`, only for `backend` and `postgres`.

Reply:
`.env` is default file
https://docs.docker.com/compose/environment-variables/#the-env-file

Reply:
Ah, ok, I see. Does it mean that all `env_file` options are not necessary? I think it's ok to have `env_file` even if it's not necessary, but I don't understand why we have `env_file` for `backend` and `postgres`, but not for `frontend`.

Reply:
Yes, it's not necessary. I didn't know that `.env` is default env file and forgot it to add `env_file` to frontend service)
How will be better, to dele `env-file` or add it to frontend service?

Reply:
I would delete it and just write a comment somewhere saying that `.env` is the default `env_file`. But also please check one thing: if we delete `env_file` and specify `--env-file .env.local`, will `.env` still be taken into account?

Reply:
Oh, look what I've found! `.env` is default, but it's applied only for compose file, and it will no forward environment inside service. So we have to keep `env-file` section in compose file for backend and postgres.

Also! passing `--env-file` seems like changing only compose file variables, so docker will no forward env inside service, same as with default file.

The decision can be specify two files in compose file:
```yaml
env_file:
      - .env
      - .env.local
```
The second file overrides env vars from first one.

What we have to do: is return back `environment` sections (and keep `env-file`), than it will forward variables inside service and allow to override them with `--env-file`


How docker builds config:
Content of `.env.local`:
```env
POSTGRES_PORT=5431
POSTGRES_DB="local-ednadb"

EDNA_API_SERVE_DOCS=false

FRONTEND_PORT=8085
```

Without `env-file` in compose file
```yaml
docker-compose --env-file=.env.local  config                                                                                                                      0 (0.527s) < 16:50:20
networks:
  edna-network: {}
services:
  backend:
    command: --config /config.yaml
    depends_on:
      postgres:
        condition: service_started
    image: ghcr.io/serokell/edna-backend:latest
    networks:
      edna-network: null
    ports:
    - published: 9000
      target: 9000
    restart: always
    volumes:
    - /Users/worm2fed/workspace/serokell/edna/deployment/config.yaml:/config.yaml:rw
  frontend:
    depends_on:
      backend:
        condition: service_started
    image: ghcr.io/serokell/edna-frontend:latest
    networks:
      edna-network: null
    ports:
    - published: 8085
      target: 80
    restart: always
  postgres:
    container_name: postgres
    image: postgres
    networks:
      edna-network: null
    ports:
    - published: 5431
      target: 5432
    restart: always
    volumes:
    - edna-db-data:/var/lib/postgresql/data:rw
version: '3.7'
volumes:
  edna-db-data: {}
 ```
 
 When I put it back:
 ```yaml
 networks:
  edna-network: {}
services:
  backend:
    command: --config /config.yaml
    depends_on:
      postgres:
        condition: service_started
    environment:
      EDNA_API_LISTEN_ADDR: '*:9000'
      EDNA_API_SERVE_DOCS: "true"
      EDNA_DB_DEBUG: "false"
      EDNA_DB_INITIALISATION_INIT_SCRIPT: /init.sql
      EDNA_DB_INITIALISATION_MODE: enable
      EDNA_DB_MAX_CONNECTIONS: '200'
      EDNA_LOGGING: dev
      FRONTEND_PORT: '8080'
      POSTGRES_DB: ednadb
      POSTGRES_HOST: postgres
      POSTGRES_PASSWORD: ednapass
      POSTGRES_PORT: '5432'
      POSTGRES_USER: ednauser
    image: ghcr.io/serokell/edna-backend:latest
    networks:
      edna-network: null
    ports:
    - published: 9000
      target: 9000
    restart: always
    volumes:
    - /Users/worm2fed/workspace/serokell/edna/deployment/config.yaml:/config.yaml:rw
  frontend:
    depends_on:
      backend:
        condition: service_started
    image: ghcr.io/serokell/edna-frontend:latest
    networks:
      edna-network: null
    ports:
    - published: 8085
      target: 80
    restart: always
  postgres:
    container_name: postgres
    environment:
      EDNA_API_LISTEN_ADDR: '*:9000'
      EDNA_API_SERVE_DOCS: "true"
      EDNA_DB_DEBUG: "false"
      EDNA_DB_INITIALISATION_INIT_SCRIPT: /init.sql
      EDNA_DB_INITIALISATION_MODE: enable
      EDNA_DB_MAX_CONNECTIONS: '200'
      EDNA_LOGGING: dev
      FRONTEND_PORT: '8080'
      POSTGRES_DB: ednadb
      POSTGRES_HOST: postgres
      POSTGRES_PASSWORD: ednapass
      POSTGRES_PORT: '5432'
      POSTGRES_USER: ednauser
    image: postgres
    networks:
      edna-network: null
    ports:
    - published: 5431
      target: 5432
    restart: always
    volumes:
    - edna-db-data:/var/lib/postgresql/data:rw
version: '3.7'
volumes:
  edna-db-data: {}
```

With two `env-file`s in compose file
```yaml
networks:
  edna-network: {}
services:
  backend:
    command: --config /config.yaml
    depends_on:
      postgres:
        condition: service_started
    environment:
      EDNA_API_LISTEN_ADDR: '*:9000'
      EDNA_API_SERVE_DOCS: "false"
      EDNA_DB_DEBUG: "false"
      EDNA_DB_INITIALISATION_INIT_SCRIPT: /init.sql
      EDNA_DB_INITIALISATION_MODE: enable
      EDNA_DB_MAX_CONNECTIONS: '200'
      EDNA_LOGGING: dev
      FRONTEND_PORT: '8085'
      POSTGRES_DB: local-ednadb
      POSTGRES_HOST: postgres
      POSTGRES_PASSWORD: ednapass
      POSTGRES_PORT: '5431'
      POSTGRES_USER: ednauser
    image: ghcr.io/serokell/edna-backend:latest
    networks:
      edna-network: null
    ports:
    - published: 9000
      target: 9000
    restart: always
    volumes:
    - /Users/worm2fed/workspace/serokell/edna/deployment/config.yaml:/config.yaml:rw
  frontend:
    depends_on:
      backend:
        condition: service_started
    image: ghcr.io/serokell/edna-frontend:latest
    networks:
      edna-network: null
    ports:
    - published: 8080
      target: 80
    restart: always
  postgres:
    container_name: postgres
    environment:
      EDNA_API_LISTEN_ADDR: '*:9000'
      EDNA_API_SERVE_DOCS: "false"
      EDNA_DB_DEBUG: "false"
      EDNA_DB_INITIALISATION_INIT_SCRIPT: /init.sql
      EDNA_DB_INITIALISATION_MODE: enable
      EDNA_DB_MAX_CONNECTIONS: '200'
      EDNA_LOGGING: dev
      FRONTEND_PORT: '8085'
      POSTGRES_DB: local-ednadb
      POSTGRES_HOST: postgres
      POSTGRES_PASSWORD: ednapass
      POSTGRES_PORT: '5431'
      POSTGRES_USER: ednauser
    image: postgres
    networks:
      edna-network: null
    ports:
    - published: 5432
      target: 5432
    restart: always
    volumes:
    - edna-db-data:/var/lib/postgresql/data:rw
version: '3.7'
volumes:
  edna-db-data: {}
```

Reply:
so ironic, that my first solution with compose file was correct, but I couldn't explain some points of it)
the good news is that we found a bug and learn more about docker

Reply:
Ok, now we are back to my previous question: why do we set `env_file` for `postgres` and `backend`, but don't set it for `frontend`?

Reply:
I've pushed one commit with additional comment.

Reply:
> Ok, now we are back to my previous question: why do we set `env_file` for `postgres` and `backend`, but don't set it for `frontend`?

Because we want to forward variables inside backend and postgres. And we have no env vars inside frontend, so we do not need it.
When we will want to change listen port of backend and forward it inside frontend image we also will have to add `env_file` section to frontend service (and write some logic to process it)